packages/react-onegraph/src/index.js

```diff
@@ -1,90 +1,97 @@
-import React, {Component, createContext} from 'react';
-import OneGraphAuth from 'onegraph-auth';
+import React, {Component, createContext} from 'react'
+import OneGraphAuth from 'onegraph-auth'
 
-const AuthContext = createContext();
+const AuthContext = createContext()
 
 class AuthProvider extends Component {
   state = {
     auth: null,
     status: {},
     headers: {},
-  };
+  }
 
   componentDidMount() {
-    const auth = new OneGraphAuth({
-      appId: this.props.appId,
-    });
+    const auth = this.getAuth()
 
-    auth.servicesStatus().then(status =>
+    auth.servicesStatus().then((status) =>
       this.setState({
         headers: auth.authHeaders(),
         status: Object.keys(status).reduce((out, service) => {
-          out[service] = status[service].isLoggedIn;
-          return out;
+          out[service] = status[service].isLoggedIn
+          return out
         }, {}),
         auth,
       }),
-    );
+    )
+  }
+
+  getAuth = () => {
```

Comment:
Can you move this logic into the state initialization. I think you should be able to do:

```js
state = {
  auth: this.props.auth || new OneGraphAuth({appId: this.props.appId})
 ...
}
```

If that doesn't work, then putting in the constructor would also be fine (it's not side-effecting, so it doesn't need to go in `componentDidMount`).